### PR TITLE
Test whether pod startup latency is good enough.

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

**Comment by wojtek-t** [comment]
cc @fgrzadkowski

File: test/e2e/density.go [hunk 1]
```diff
@@ -148,7 +148,6 @@ var _ = Describe("Density", func() {
 		expectNoError(writePerfData(c, fmt.Sprintf(testContext.OutputDir+"/%s", uuid), "after"))
 
 		// Verify latency metrics
-		// TODO: We should reset metrics before the test. Currently previous tests influence latency metrics.
```
Comment (fgrzadkowski) [inline test/e2e/density.go]: Why remove this?

File: test/e2e/density.go [hunk 2]
```diff
@@ -386,6 +385,16 @@ var _ = Describe("Density", func() {
 				printLatencies(schedToWatchLag, "worst scheduled-to-end total latencies")
 				printLatencies(e2eLag, "worst e2e total latencies")
 
+				// Test whether e2e startup time is acceptable.
```
Comment (fgrzadkowski) [inline test/e2e/density.go]: s/e2e startup/e2e pod startup

File: test/e2e/density.go (hunk 1 above)
Comment (wojtek-t) [inline test/e2e/density.go]: It's already done :)

File: test/e2e/density.go (hunk 2 above)
Comment (wojtek-t) [inline test/e2e/density.go]: Done.

**Comment by fgrzadkowski** [comment]
LGTM apart from two small comments. I'll add lgtm label once they are addressed.

**Comment by wojtek-t** [comment]
Done - PTAL

**Comment by pmorie** [comment]
@wojtekt @fgrzadkowski What is the reasoning about only running the latency test in the 30 pod per minion case?  How do you feel about splitting out the latency test into a separate E2E?  It seems orthogonal to the density aspect of the test.

**Comment by wojtek-t** [comment]
@pmorie - The most important thing is that we would like to run "latency test" in the full cluster (ie where there are ~30 pods per node are running) as this is the worst situation from the performance point of view.
I agree that it might be useful to split this into a separate test at some point, but currently our e2e tests take over 2 hours, and I would like them to be running as often as possible (to detect regressions).
So Yes - I agree that it might be useful at some point, but I think we're not there yet.

**Comment by pmorie** [comment]
@wojtek-t I wound up factoring out a distinct test for it, but I understand the rationale you described.  Agree that ideally this test should run in a primed cluster.  In this case, I can reliably reproduce the failure on a single node with no pods running with the extracted test.  I will leave in the ability to run the latency test in a cluster primed by the density test.

**Comment by wojtek-t** [comment]
@pmorie - thanks - SGTM. Can you please cc me to your PR when it will be ready?

**Comment by pmorie** [comment]
@wojtek-t yep :)

**Pull request merged**